python/lsst/pipe/tasks/processBrightStars.py

```diff
@@ -249,109 +338,191 @@ def applySkyCorr(self, calexp, skyCorr):
+        At that point, it should be replaced by a call to the Exposure method
+        ``getCutout``, which will handle edge cases automatically.
+        """
+        corner = Point2I(np.array(coordPix) - np.array(stampSize) / 2)
+        dimensions = Extent2I(stampSize)
+        stampBBox = Box2I(corner, dimensions)
+        overlapBBox = Box2I(stampBBox)
+        overlapBBox.clip(inputExposure.getBBox())
+        if overlapBBox.getArea() > 0:
+            # Create full-sized stamp with pixels initially flagged as NO_DATA.
+            stamp = ExposureF(bbox=stampBBox)
+            stamp.image[:] = np.nan
+            stamp.mask.set(inputExposure.mask.getPlaneBitMask("NO_DATA"))
+            # Restore pixels which overlap the input exposure.
+            inputMI = inputExposure.maskedImage
+            overlap = inputMI.Factory(inputMI, overlapBBox)
+            stamp.maskedImage[overlapBBox] = overlap
+            # Set detector and WCS.
+            stamp.setDetector(inputExposure.getDetector())
+            stamp.setWcs(inputExposure.getWcs())
+        else:
+            stamp = None
+        return stamp
+
+    def _replaceSecondaryFootprints(self, stamp, coordPix, objectId, find="DETECTED", replace="BAD"):
+        """Replace all secondary footprints in a stamp with another mask flag.
+
+        This method identifies all secondary footprints in a stamp as those
+        whose ``find`` footprints do not overlap the given pixel coordinates.
+        If then sets these secondary footprints to the ``replace`` flag.
+
+        Parameters
+        ----------
+        stamp : `~lsst.afw.image.ExposureF`
+            The postage stamp to modify.
+        coordPix : `~lsst.geom.Point2D`
+            The pixel coordinates of the central primary object.
+        objectId : `int`
+            The unique identifier of the central primary object.
+        find : `str`, optional
+            The mask plane to use to identify secondary footprints.
+        replace : `str`, optional
+            The mask plane to set secondary footprints to.
+
+        Notes
+        -----
+        This method modifies the input ``stamp`` in-place.
+        """
+        # Find a FootprintSet given an Image and a threshold.
+        detThreshold = Threshold(stamp.mask.getPlaneBitMask(find), Threshold.BITMASK)
+        footprintSet = FootprintSet(stamp.mask, detThreshold)
+        allFootprints = footprintSet.getFootprints()
+        # Identify secondary objects (i.e., not the central primary object).
+        secondaryFootprints = []
+        for footprint in allFootprints:
+            if not footprint.contains(Point2I(coordPix)):
+                secondaryFootprints.append(footprint)
+        # Set secondary object footprints to BAD.
+        if (numPrimaryFootprints := len(allFootprints) - len(secondaryFootprints)) != 1:
+            self.log.warning(
+                "Could not uniquely identify central %s footprint for star %s; "
```

Comment:
That's correct. This method attempts to identify all secondary sources in the frame and masks them. If the primary source _is_ identified, those pixels are not masked. If the primary source is _not_ identified (e.g., for a low-S/N source), processing can still continue and those pixels may be stacked when producing an extended PSF.

In an attempt to help make this clearer, we have changed the logic in the `if` block which definines the `numPrimaryFootprints` variable from `!= 1` to `== 0` (as only values of 0 or 1 will ever be returned by `numPrimaryFootprints`). We have also added this in-line comment block:

```
# Note: the value of numPrimaryFootprints can only by 0 or 1. If it is
# 0, then the primary object was not found overlapping a footprint.
# This can occur for low-S/N stars, for example. Processing can still
# continue beyond this point in an attempt to utilize this faint flux.
```

Reply:
Okay. In that case I'm not sure if this should be a warning rather than info, because it sounds like it's not entirely unusual. On the other hand, if it's something that would make subsequent data products potentially unreliable then how would a user know (i.e. is there a catalog you could add a flag to)?

Reply:
Changed to `info` level logging. Will keep a close eye on log outputs in the coming testing phases.